nodes/src/nodes/approval/approval_manager.py

```
@@ -0,0 +1,218 @@
+            'created_at': time.monotonic(),
+            'timeout_seconds': self._timeout_seconds,
+            'timeout_action': self._timeout_action,
+            'reviewer': None,
+            'review_comment': None,
+            'reviewed_at': None,
+        }
+
+        with self._lock:
+            pending_count = sum(1 for r in self._requests.values() if r['status'] == 'pending')
+            if pending_count >= MAX_PENDING_REQUESTS:
+                raise ValueError(f'Maximum pending requests ({MAX_PENDING_REQUESTS}) reached')
```

Ruff (0.15.7)

[warning] Avoid specifying long messages outside the exception class

(TRY003)